### DISCUSSION TITLE
Welcome to SHM-MOD Discussions!

### DISCUSSION THREAD

**Discussion by ITDev93** [ITDev93/SHM-MOD#1] [Announcements]
## 👋 Hi there!
  I hope you're fine...
  Welcome to SHM-MOD Discussions, a place to connect with other members of this community. I hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  📢 **Don't forget,** not so many know about this github! So kindly, go ahead and send that tweet, post, or link it from the website to drive traffic here a bit for those who don'thave Telegram or XDA.

  To get started, comment below with an introduction of what you did, the device you have (watch and phone) and tell me about the issue or feedback or whatever you wish to share about SHM MOD.

**Comment by NickTheOnlyNeko** [comment]
Why are you keeping the App Closed Source? Is it because out of fear that Samsung might patch it out or something else??

**Reply by ITDev93** [reply]
- The app is owned by Samsung, not me, thus it's not Open Source...
- I always provide stock and mod SHM so anyone who wish to see the changes simply decompile and compare them...
- The GitHub is created for 2 reasons:
      1. An alternative to XDA, easier to read and understand...
      2. SHM MOD Companion reads bugs, updates from GitHub...

**Comment by Danas544** [comment]
Hello work watch8?

**Reply by ITDev93** [reply]
It is mentioned already:
<img width="1001" height="622" alt="image" src="https://github.com/user-attachments/assets/8f11baf7-40d8-401d-98da-dc0794cc6427" />

**Comment by F242Thom** [comment]
Hi.  Love the factt you are diligent in making bp available to usa samsung users...however, the sheer amount of your work on youtube and xda leaves me confused as 2 where I should start.  Tried following watch 4 instructions...got lost at end ... Watch  is galaxy 6 with ui 6.  Phone is 10e with ui 4..1 so should work...I know the issue is my general inexperience with xda and github and frankly I hardly ever use my phone so using it to find stuff ... Are there links in the comments at youtube ?

**Reply by ITDev93** [reply]
Update the companion, and follow the video instructions in guide, despite the video is old and not matching the new ui, I have added short videos that covers the new ui...

**Reply by F242Thom** [reply]
Got as far as connecting phone and watch via companion but then 20 seconds and it timed out.  Trying to get the watch companion onto the watch...the "what do you want to do" is not available (greyed out).  Geez...I ain't stupid, I'm a nuclear engineer for gosake.  But this is driving me nuts.  Tried another route since have PC.  But Watch Companion App APK ... where?

**Comment by F242Thom** [comment]
Perseverance was required.  The watch failed to connect with PC.   Watch time out was 20 seconds with phone...so I tried pixel tablet and finally got stable connection.  Lots of frustration...like all software.

Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, Oct 4, 2025 at 10:14, ***@***.***> wrote:   


Update the companion, and follow the video instructions in guide, despite the video is old and not matching the new ui, I have added short videos that covers the new ui...

**Reply by ITDev93** [reply]
When you initiate the connection, make sure the watch screen is on all the times, some people recommend turning off bluetooth to force wireless to remain on...

This is due to how aggressive battrey optimization is on the watch, as soon watch screen goes off, it enters dose mode, nothing runs but essential services, even wireless debugging, since no connection initiated, it is by default suspended and not responding...

Additionally, avoid using VPN if used, and try your own phone network (tethering or hotspot) to ensure a fast connectivity with least interruptions...

Once you have the watch companion installed, connecting in the future should be smoother as the phone app wakes up the watch for connection...

An expansion of shm mod companion, if you need an adb tool on the phone to manage your watch is geminiman WearOS manager...
